plugins/modules/azuread_group.py

```diff
@@ -507,71 +532,57 @@ def compare_groups(current, new):
-    for item in new_keys_to_remove:
-        if item in new:
-            new.pop(item)
     if current != new:
         return dict(before=current, after=new)
 
 
+def build_group_from_playbook(params):
+    GROUP_PARAMS = ["display_name", "description", "group_types", "mail_enabled",
+                    "mail_nickname", "security_enabled", "owners", "members"]
+    group = {}
+    for param in GROUP_PARAMS:
+        group[param] = params[param]
+    return snake_dict_to_camel_dict(group)
+
+
 def main():
     module = setup_module_object()
+    azuread_iface = AzureActiveDirectoryInterface(module)
 
     if not HAS_DEPS:
         module.fail_json(msg="module requires requests and requests-oauthlib")
 
+    changed = False
+    diff = None
+
     state = module.params['state']
     name = module.params['display_name']
-    owners = module.params['owners']
     enforce_owners = module.params['enforce_owners']
-
-    members = module.params['members']
     enforce_members = module.params['enforce_members']
+    new_group = build_group_from_playbook(module.params)
+    current_group = azuread_iface.build_group_from_api(name)
 
-    azuread_iface = AzureActiveDirectoryInterface(module)
-    group = azuread_iface.get_group(name)
-
-    changed = False
-    diff = None
     if state == 'present':
-        new_group = build_group_from_params(module.params)
-        if group is None:
+        if current_group is None:
             group = azuread_iface.create_group(new_group)
             changed = True
         else:
-
-            diff = compare_groups(group.copy(), new_group.copy())
+            current_group_id = current_group.get("id")
+            diff = compare_groups(current_group.copy(), new_group.copy())
             if diff is not None:
-                azuread_iface.update_group(group.get("id"), diff["after"])
-                changed = True
-
-            current_owners = azuread_iface.get_owners_id(group.get("id"))
-            if current_owners != owners:
-                owners_changed = azuread_iface.converge_owners(group.get("id"), current_owners, owners, enforce_owners)
-                if owners_changed:
-                    changed = True
-
-            current_members = azuread_iface.get_members_id(group.get("id"))
-            if current_members != members:
-                members_changed = azuread_iface.converge_members(group.get("id"), current_members, members,
-                                                                 enforce_members)
-                if members_changed:
-                    changed = True
-
-        group = azuread_iface.get_group(name)
-        group["owners"] = azuread_iface.get_owners(group.get("id"))
-        group["members"] = azuread_iface.get_members(group.get("id"))
+                changed = azuread_iface.converge_group(current_group_id, diff.copy(), enforce_owners, enforce_members)
+        group = azuread_iface.build_group_from_api(name)
```

Comment:
Ce qui est cool maintenant c'est que tu peux facilement supporter le check_mode ansible :

```suggestion
                if module.check_mode:
                    module.exit_json(changed=True, group=diff["after"], diff=diff)
                changed = azuread_iface.converge_group(current_group_id, diff.copy(), enforce_owners, enforce_members)
        group = azuread_iface.build_group_from_api(name)
```

A tester !
En lançant Ansible avec l'option `--check`